src/processors/processPostRequest.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+import { isValidObject } from '@dvsa/cvs-type-definitions/schema-validator';
+import logger from '../util/logger';
+import { generateNewNumber, NumberTypes } from '../services/testNumber';
+import { identifySchema } from '../validators/post';
+import { UserDetails } from '../services/user';
+import { HttpMethod, RecordCompleteness, VehicleType } from '../util/enum';
+import { TechrecordGet, TechrecordPut } from '../models/post';
+
+export const processPostRequest = async (input: TechrecordPut, userDetails: UserDetails) => {
+  const request: TechrecordPut = await flattenArrays(input);
+  logger.info('processing request');
+  // helper method for handler
+  const systemNumber = await generateNewNumber(NumberTypes.SystemNumber);
+  if (request.techRecord_vehicleType !== 'trl' && !request.primaryVrm) {
+    request.primaryVrm = await generateNewNumber(NumberTypes.ZNumber);
+  }
+  if (request.techRecord_vehicleType === 'trl' && !request.trailerId && (request.techRecord_euVehicleCategory === 'o1' || request.techRecord_euVehicleCategory === 'o2')) {
+    request.trailerId = await generateNewNumber(NumberTypes.TNumber);
+  }
+  if (request.techRecord_vehicleType === 'trl' && !request.trailerId) {
+    request.trailerId = await generateNewNumber(NumberTypes.TrailerId);
+  }
+  if ('techRecord_recordCompleteness' in request) {
+    request.techRecord_recordCompleteness = computeRecordCompleteness(request);
+  }
+  if ('systemNumber' in request) {
+    request.systemNumber = systemNumber;
+  }
+  if ('createdTimestamp' in request) {
+    request.createdTimestamp = new Date().toISOString();
+  }
+  if ('vin' in request && request.vin) {
+    request.partialVin = request.vin.length < 6 ? request.vin : request.vin.substring(request.vin.length - 6);
+  }
+  if ('techRecord_createdByName' in request) {
+    request.techRecord_createdByName = userDetails.username;
+  }
+  if ('techRecord_createdById' in request) {
+    request.techRecord_createdById = userDetails.msOid;
+  }
```

Comment:
I'm confused... why would the property be `in` the request.

I'd say done something like:

```ts
(request as TechrecordGet).systemNumber = systemNumber
(request as TechrecordGet).createdTimeStamp= systemNumber
```

Or you even could give `request` the type of `TechrecordGet` on line 16

Reply:
Some of those values don't seem to be accessible inside of that object in the example of primaryVrm, it's causing an error due to it not being present on a TRL & I don't think we should be doing a mix of both

> Property 'primaryVrm' does not exist on type 'TechrecordGet'.   Property 'primaryVrm' does not exist on type 'GETTRLTechnicalRecordV3Testable'.